data/filter/FieldFilter.js

```diff
@@ -96,48 +96,66 @@ export class FieldFilter extends Filter {
         switch (op) {
             case '=':
                 return r => {
+                    if (isNil(r.committedData)) return true; // Ignore (do not filter out) record if not committed
```

Comment:
I think we need to change these conditions to `if (store && isNil(r.committedData))` - filters are supposed to be able to work against anonymous objects as well as StoreRecords (see [getTestFn docs](https://github.com/xh/hoist-react/blob/develop/data/filter/Filter.js#L28-L30)). As part of that change, we may as well put the condition into a named function e.g. `isUncommittedRecord(r)`.

Reply:
re `store`: yes, of course, thanks for pointing that out.  I have made the fix and moved it all into a new local function called `mustIgnore`.